modules/axhal/src/platform/loongarch64_qemu_virt/console.rs

```diff
@@ -0,0 +1,44 @@
+use crate::mem::phys_to_virt;
+use kspin::SpinNoIrq;
+use lazyinit::LazyInit;
+use memory_addr::PhysAddr;
+use ns16550a::Uart;
+
+const UART_BASE: PhysAddr = pa!(axconfig::devices::UART_PADDR);
+
+static UART: LazyInit<SpinNoIrq<Uart>> = LazyInit::new();
+
+/// Writes bytes to the console from input u8 slice.
+pub fn write_bytes(bytes: &[u8]) {
+    let uart = UART.lock();
```

Comment:
If there are too many bytes, the lock may be held for a long time.

Try to slice the bytes and acquire the lock multiple times.